### ISSUE TITLE
LexProps: don't set header flag for empty section

### ISSUE THREAD

**Issue by zufuliu** [ScintillaOrg/lexilla#96]
To prevent `FoldAction::Toggle` failure and avoid showing fold display text after toggle folds.

```diff
diff --git a/lexers/LexProps.cxx b/lexers/LexProps.cxx
index 7dadb2c6..b869659a 100644
--- a/lexers/LexProps.cxx
+++ b/lexers/LexProps.cxx
@@ -154,6 +154,10 @@ static void FoldPropsDoc(Sci_PositionU startPos, Sci_Position length, int, WordL
 
 			if (headerPoint) {
 				lev |= SC_FOLDLEVELHEADERFLAG;
+				if (levelPrevious & SC_FOLDLEVELHEADERFLAG) {
+					// empty section
+					styler.SetLevel(lineCurrent - 1, SC_FOLDLEVELBASE);
+				}
 			}
 			if (lev != styler.LevelAt(lineCurrent)) {
 				styler.SetLevel(lineCurrent, lev);
```

**Comment by nyamatongwe** [comment]
Is there an example file?

**Comment by zufuliu** [comment]
An example, toggle folds not fails in SciTE (not sure why), fold display text can't be tested in SciTE.
```ini
; comment
[empty section]
[normal section]
key=value
```

**Comment by zufuliu** [comment]
Full code for `FoldPropsDoc()`, moved `levelPrevious` out from the loop.
[FoldPropsDoc.zip](https://github.com/ScintillaOrg/lexilla/files/9185908/FoldPropsDoc.zip)

**Comment by nyamatongwe** [comment]
`FoldPropsDoc.zip` won't compile as `lev` is used in the epilogue but its declaration is now in an inner scope. I expect it would be better as:
~~~c++
	int lev = levelPrevious & SC_FOLDLEVELNUMBERMASK;
	if (levelPrevious & SC_FOLDLEVELHEADERFLAG) {
		lev = SC_FOLDLEVELBASE + 1;
	}
~~~
Or with another name to avoid shadowing. The inner declaration of `lev` could also be initialised to `levelPrevious & SC_FOLDLEVELNUMBERMASK` to avoid uninitialized variable warnings.

**Comment by zufuliu** [comment]
It seems last `styler.SetLevel(lineCurrent, lev | (flagsNext & ~SC_FOLDLEVELNUMBERMASK));` can be removed (change inner code to use `if (atEOL || i == (endPos - 1)) {`), not sure what it want archive.

**Comment by nyamatongwe** [comment]
This pattern is used to avoid transient changes (and potential unfolding) due to not running the full logic past the end. It may or may not be needed in this case.

**Comment by zufuliu** [comment]
Changed to following
```cpp
int level = levelPrevious & SC_FOLDLEVELNUMBERMASK;
if (levelPrevious & SC_FOLDLEVELHEADERFLAG) {
	level += 1;
}
```

[FoldPropsDoc-0727.zip](https://github.com/ScintillaOrg/lexilla/files/9193887/FoldPropsDoc-0727.zip)

**Comment by nyamatongwe** [comment]
Included in 5.1.9.
